tools/tsp-client/src/commands.ts

```diff
@@ -405,6 +405,61 @@ export async function convertCommand(argv: any): Promise<void> {
   }
 }
 
+export async function generateConfigFilesCommand(argv: any) {
+  const outputDir = argv["output-dir"];
+  const repoRoot = await getRepoRoot(outputDir);
+  const packageJsonPath = normalizePath(resolve(argv["package-json"]));
+  const overridePath = argv["overrides"] ?? undefined;
+
+  if (packageJsonPath === undefined || !(await doesFileExist(packageJsonPath))) {
+    throw new Error(`package.json not found in: ${packageJsonPath ?? "[Not Specified]"}`);
+  }
+  Logger.info("Generating emitter-package.json file...");
+  const content = await readFile(packageJsonPath);
+  const packageJson: Record<string, any> = JSON.parse(content.toString());
+  const emitterPackageJson: Record<string, any> = {
+    name: "dist/src/index.js",
+    dependencies: {},
+  };
+
+  let overrideJson: Record<string, any> = {};
+  if (overridePath) {
+    overrideJson = JSON.parse((await readFile(overridePath)).toString()) ?? {};
+  }
+
+  // Add emitter as dependency
+  emitterPackageJson["dependencies"][packageJson["name"]] =
+    overrideJson[packageJson["name"]] ?? packageJson["version"];
+
+  delete overrideJson[packageJson["name"]];
+  const devDependencies: Record<string, any> = {};
+
+  const possiblyPinnedPackages =
+    packageJson["azure-sdk/emitter-package-json-pinning"] ?? packageJson["peerDependencies"];
```

Comment:
You'll be processing `possiblyPinnedPackages` as a string array, so you'll want to pull the keys from `peerDependencies`.

`azure-sdk/emitter-package-json-pinning` will already be a string array.

Reply:
In powershell, we use the Keys property:
`$packageJson.peerDependencies.Keys`

Does javascript enumerate the keys of an object with a `for` loop?